SignalMessaging/utils/UIImage+OWS.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+extension UIImage {
+
+    private func image(with view: UIView) -> UIImage? {
+        UIGraphicsBeginImageContextWithOptions(view.bounds.size, view.isOpaque, 0.0)
+        defer { UIGraphicsEndImageContext() }
+
+        guard let context = UIGraphicsGetCurrentContext() else {
+            return nil
+        }
+
+        view.layer.render(in: context)
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        return image
+    }
+
+    @objc
+    public func templated(color: UIColor) -> UIImage? {
```

Comment:
I'd propose `asTintedView`?

Reply:
Yeah, that's better. I struggled with this one. =)